### PR TITLE
Do not wait the end of a reset to return an update

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
return the update endpoint without having to wait for a reset to finish running. 

## Important files to review

`ConnectionManagerWorkflow.java` -> add a new signal
`WorkflowState.java` -> Add a new state that allow to continue the new run without waiting for the scheduling
`ConnectionManagerWorkflowImpl.java` -> User the new state
`TemporalClient.java` -> Remove the synchronous reset and call the new signal if needed. 

Address #16178

### CODE REVIEW

**Comment by benmoriceau** [comment]
Hmm I don't know why the build didn't run.

**Comment by benmoriceau** [comment]
I tested the Basic and Advance Acceptance Test locally. They passed

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by benmoriceau** [comment]
I relaunched the test, I think that the error is transient

File: airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java [hunk 1]
```diff
@@ -642,7 +643,7 @@ void testUpdateConnectionWithUpdatedSchemaLegacy() throws JsonValidationExceptio
     when(configRepository.getAllStreamsForConnection(expected.getConnectionId())).thenReturn(connectionStreams);
 
     final ManualOperationResult successfulResult = ManualOperationResult.builder().jobId(Optional.empty()).failingReason(Optional.empty()).build();
-    when(eventRunner.synchronousResetConnection(any(), any())).thenReturn(successfulResult);
+    when(eventRunner.resetConnection(any(), any(), anyBoolean())).thenReturn(successfulResult);
     when(eventRunner.startNewManualSync(any())).thenReturn(successfulResult);
```
Comment (gosusnp) [inline airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java:647]: nit: Comment also applies for the other tests: I think we no longer need this mock, can we remove it?

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java [hunk 2]
```diff
@@ -492,6 +492,18 @@ public void resetConnection() {
     }
   }
 
+  @Override
+  public void resetConnectionAndSkipNextScheduling() {
+    if (workflowState.isDoneWaiting()) {
+      workflowState.setCancelledForReset(true);
```
Comment (gosusnp) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:498]: For my understanding, `setCancelledForReset` means that we cancel the current run and start a reset?

**Review by gosusnp — APPROVED** [review]

File: airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java (hunk 1 above)
Comment (benmoriceau) [reply airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java:647]: @gosusnp The build is green, I'll merge/deploy and address your comments in another PR

**Pull request merged**